### PR TITLE
[DA-2427] Update report state on consent removal

### PR DESCRIPTION
**Pull request by j-kanuch**
## Description of changes/additions
This PR creates a cron job for the updating the participant report state when gRoR and primary consent is removed. This PR modifies the query in `get_unconsented_gror_since_date()` to remove the date constraint. Methods that were previously specified for GEM only were renamed to be more generic. Logic was updated to check for the existence of consents to prevent value errors.

## Tests
- [x] unit tests

### CODE REVIEW

**Review by dtharpeuno — APPROVED** [review]

File: rdr_service/dao/genomics_dao.py [hunk 1]
```diff
@@ -580,13 +580,18 @@ def get_gem_consent_removal_date(self, member):
...
         # Earliest date between GROR or Primary if both,
         withdraw_dates = []
-        if consent_status.consentForGenomicsROR != QuestionnaireStatus.SUBMITTED:
+        if consent_status.consentForGenomicsROR != QuestionnaireStatus.SUBMITTED and \
+            consent_status.consentForGenomicsRORAuthored:
             withdraw_dates.append(consent_status.consentForGenomicsRORAuthored)
 
-        if consent_status.consentForStudyEnrollment != QuestionnaireStatus.SUBMITTED:
+        if consent_status.consentForStudyEnrollment != QuestionnaireStatus.SUBMITTED and \
+            consent_status.consentForStudyEnrollmentAuthored:
```
Comment (kenny-skaggs) [inline rdr_service/dao/genomics_dao.py:590]: Is `consentForStudyEnrollment` ever anything but `SUBMITTED`? I think what we've been needing to do is remove the participant summary altogether if we need to remove their Primary consent status.

File: rdr_service/offline/main.py [hunk 2]
```diff
@@ -938,6 +945,12 @@ def _build_pipeline_app():
         view_func=genomic_gem_a3_workflow,
         methods=["GET"]
     )
+    offline_app.add_url_rule(
+        OFFLINE_PREFIX + "GenomicUpdateReportStateForConsentRemoval",
+        endpoint="update_report_state_for_consent_removal",
+        view_func=update_report_state_for_consent_removal,
+        methods=["GET"]
+    )
```
Comment (kenny-skaggs) [inline rdr_service/offline/main.py:953]: Should the cron job be scheduled (like in cron_default.yaml), or is it intentional to leave it unscheduled for now?

Comment (j-kanuch) [reply rdr_service/offline/main.py:953]: Yes it should be scheduled. I completely forgot to add it to the cron schedule, thanks!

File: rdr_service/dao/genomics_dao.py (hunk 1 above)
Comment (j-kanuch) [reply rdr_service/dao/genomics_dao.py:590]: I didn't realize that we remove the records. If that is the case, then this is really just a safety precaution. I will add a check for the withdrawal status as well.

**Review by dtharpeuno — APPROVED** [review]

**Review by kenny-skaggs — APPROVED** [review]

**Pull request merged**